gym/core.py

```diff
@@ -58,7 +63,7 @@ def step(self, action):
         raise NotImplementedError
 
     @abstractmethod
-    def reset(self):
+    def reset(self, seed: Optional[int] = None):
```

Comment:
```suggestion
    def reset(self, *, seed: Optional[int] = None):
```

Mark `seed` as a keyword-only argument. Same modification for all subclasses.